diff --git a/crates/eww/src/widgets/widget_definitions.rs b/crates/eww/src/widgets/widget_definitions.rs
index 8284d0f..92c526e 100644
--- a/crates/eww/src/widgets/widget_definitions.rs
+++ b/crates/eww/src/widgets/widget_definitions.rs
@@ -953,9 +953,20 @@ fn build_gtk_event_box(bargs: &mut BuilderArgs) -> Result<gtk::EventBox> {
                     3 => run_command(timeout, &onrightclick, &[] as &[&str]),
                     _ => {},
                 }
+                glib::Propagation::Proceed
+            }));
+        },
+        // @prop timeout - timeout of the command. Default: "200ms"
+        // @prop onkeyrelease - command to run when keyboard is clicked. The placeholder `{}` will be replaced by the value
+        prop(timeout: as_duration = Duration::from_millis(200), onkeyrelease: as_string) {
+            gtk_widget.add_events(gdk::EventMask::KEY_RELEASE_MASK);
+            connect_signal_handler!(gtk_widget, gtk_widget.connect_key_release_event(move |_, evt| {
+                run_command(timeout, &onkeyrelease, &[evt.scancode()]);
+
                 glib::Propagation::Proceed
             }));
         }
+
     });
     Ok(gtk_widget)
 }
